include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -0,0 +1,683 @@
+       */
+      void
+      set_tape_buffer_sizes(const types::tape_buffer_sizes obufsize = 67108864,
+                            const types::tape_buffer_sizes lbufsize = 67108864,
+                            const types::tape_buffer_sizes vbufsize = 67108864,
+                            const types::tape_buffer_sizes tbufsize = 67108864);
+
+      /**
+       * Enable recording mode for a given tape. The use of this function is
+       * mandatory if the auto-differentiable number is a taped type. However,
+       * for the purpose of developing generic code, it can also be safely
+       * called for tapeless auto-differentiable numbers.
+       *
+       * The operations that take place between this function call and that
+       * of disable_record_sensitivities() are recorded to the tape and can
+       * be replayed and reevaluated as necessary.
+       *
+       * The typical set of operations to be performed during this phase are:
+       *   - Definition of some independent variables via
+       *     register_independent_variable() or
+       *     register_independent_variables(). These define the branch of
+       *     operations tracked by the tape. If the @p keep flag is set to
+       *     <tt>true</tt> then these represent precisely the point about which
+       *     the function derivatives are to be computed. If the @p keep flag is
+       *     set to <tt>false</tt> then these only represent dummy values, and
+       *     the point at which the function derivatives are to be computed must
+       *     be set by calling set_independent_variables() again.
+       *   - Extraction of a set of independent variables of auto-differentiable
+       *     type using get_sensitive_variables(). These are then tracked during
+       *     later computations.
+       *   - Defining the dependent variables via register_dependent_variable()
+       *     or register_dependent_variables(). These are the functions that
+       *     will be differentiated with respect to the independent variables.
+       *
+       * @param[in] tape_index The index of the tape to be written
+       * @param[in] overwrite_tape Express whether tapes are allowed to be
+       * overwritten. If <tt>true</tt> then any existing tape with a given
+       * @p tape_index will be destroyed and a new tape traced over it.
+       * @param[in] keep_values Determines whether the numerical values of all
+       * independent variables are recorded in the tape buffer. If true, then
+       * tape can be immediately used to perform computations after recording
+       * is complete.
+       *
+       * @note During the recording phase, no value(), gradient(), hessian(),
+       * or jacobian() operations can be performed.
+       *
+       * @note The chosen tape index must be greater than
+       * types::invalid_tape_index and less than types::max_tape_index.
+       */
+      bool
+      enable_record_sensitivities(const types::tape_index &tape_index,
```

Comment:
Have you thought about calling these functions `start_record_sensitivities()` and `stop_record_sensitivities()`? That's because a strict interpretation of the word would imply that `enable/disable` really only says that you make it possible/impossible to record, but not necessarily that it actually happens. 

I don't know whether it's necessary to make this change, but wanted to bring it up anyway.